### ISSUE TITLE
Bad Flicking on Linux Gnome

### ISSUE THREAD

**Issue by andrewauclair** [andrewauclair/ModernDocking#218]
Looks like the overlays flicker on and off really bad when dragging a window in Gnome on Arch.

**Comment by andrewauclair** [comment]
Played around with this for a while and reducing the number of repaint helps. I'm wondering if it's because I'm on an old laptop with integrated graphics. I'm sure that doesn't help.

**Comment by andrewauclair** [comment]
This will be fixed with the overlay refactor and was finally achieved by using a `BufferStrategy` instead of manually repainting the `JFrame`.

**Comment by Mino260806** [comment]
Same issue on Linux Mint 21.1. The flicking causes a huge lag.

https://github.com/user-attachments/assets/f31c18aa-59b9-464b-9644-c7601354e2a8

**Comment by andrewauclair** [comment]
Interesting. That is the exact behavior I've seen on Arch. Maybe integrated graphics is the common issue. I had found a fix for it, but it broke the windows side of Modern Docking. I'll look into some way that I can do Linux specific code.

**Comment by Mino260806** [comment]
> Maybe integrated graphics is the common issue.

Maybe it's an issue with Gnome rather than integrated graphics? The person in the SO question mentions having "nvidia drivers", so   he most likely has dedicated graphics.

Did you confirm it works in a linux computer with dedicated graphics ?


> I had found a fix for it, but it broke the windows side of Modern Docking

Why don't you put a check on the OS type and operate accordingly ?

-----
There is something else to point out, the overlay thinks the window size is bigger than it originally is, so the right and bottom handles overflow the screen (see video again). Did you also find a fix for this ?

**Comment by andrewauclair** [comment]
> Maybe it's an issue with Gnome rather than integrated graphics? The person in the SO question mentions having "nvidia drivers", so   he most likely has dedicated graphics.
> 
> Did you confirm it works in a linux computer with dedicated graphics ?

Yes. I use this at work on Gnome on CentOS and haven't encountered this issue there. 

> > I had found a fix for it, but it broke the windows side of Modern Docking
> 
> Why don't you put a check on the OS type and operate accordingly ?

That's likely what I'll try. 

> -----
> There is something else to point out, the overlay thinks the window size is bigger than it originally is, so the right and bottom handles overflow the screen (see video again). Did you also find a fix for this ?

That is intentional. The overlap frame is larger to account for drawing handles off the edge of the main frame.

**Comment by andrewauclair** [comment]
> Maybe integrated graphics is the common issue.

Good news. I finally had the time and energy to fix this. I believe I have a working solution. The common issue appears to be integrated graphics not supporting multi-buffer. Wrote some code that will work one way for multi-buffer and another when it's not available and have tested that.

> There is something else to point out, the overlay thinks the window size is bigger than it originally is, so the right and bottom handles overflow the screen (see video again). Did you also find a fix for this ?

Sorry, I just realized this is actually an issue. Looks to be specific to using the application in full screen mode. Looking into it now.

**Comment by andrewauclair** [comment]
I recently acquired a new Windows 11 laptop with integrated graphics and verified that this does not happen there.

**Comment by nroduit** [comment]
I am experiencing the same issue on Ubuntu 24.04 LTS. This makes the framework unusable on most Linux distributions.

**Comment by andrewauclair** [comment]
> I am experiencing the same issue on Ubuntu 24.04 LTS. This makes the framework unusable on most Linux distributions.

Do you have integrated graphics? The issue should be isolated to that.

**Comment by nroduit** [comment]
I only try with demo package.

**Comment by andrewauclair** [comment]
> I only try with demo package.

That's fine. I was asking what hardware you're running on, not what you were running.

**Comment by nroduit** [comment]
- **Graphic card :**   NVIDIA RTX A4000
- **Operating system**   Ubuntu 24.04.2 LTS
- **GNOME version :**   46
- **Windows system :**  X11
- **Kernel version :**      Linux 6.8.0-58-generic

**Comment by nroduit** [comment]
The last commit seems to fix the flickering problem, but with a light theme the choice of graphics colors for the separation is unsuitable.

https://github.com/user-attachments/assets/cd865ef2-a028-4977-b485-a637fb5b86e5

**Comment by andrewauclair** [comment]
> The last commit seems to fix the flickering problem, but with a light theme the choice of graphics colors for the separation is unsuitable.

The colors can be controlled through `DockingSettings`. By default, the handle colors are pulled from `TableHeader` colors in UIManager. The overlay color should default to blue, so I'm not sure what went wrong with yours.

**Comment by andrewauclair** [comment]
The flicking and transparency issues have been addressed in 1.3.1.
